### PR TITLE
Docker analysis refactoring

### PR DESCRIPTION
**Pull request by bcorrie**
To be reviewed by @jeromejaglale

### CODE REVIEW

**Review by jeromejaglale — CHANGES_REQUESTED** [review]
Added some comments on JIRA: https://ireceptor.atlassian.net/browse/IR-2483

File: app/Agave.php [hunk 1]
```diff
@@ -111,6 +112,41 @@ public function renewToken($url, $refresh_token, $api_key, $api_secret)
         return $response;
     }
 
+    public function updateAppTemplates()
+    {
+        $this->appTemplates = [];
+        $app_directories = config('services.agave.app_directories');
```
[the remark this answers is not in the document]
Comment (bcorrie) [reply app/Agave.php:118]: No, I don't think so... I though of doing that, but I thought it safer to be explicit. That way we are sure that what we are running through the Gateway is intentional and precise. With a config we can have development apps that are not in production, can temporarily add/remove apps easily, without touching the files/directories...

In addition, with singularity, from a security standpoint, I think we want to be very careful about what we run as Apps. So scanning a directory could be risky. The way I have it, the Gateway config controls which apps are run, and the Gateway separately controls which singularity images are used (they aren't bundled with the app). This makes it very difficult for someone to add a rogue singularity image or a rogue app that does something bad.

File: app/Jobs/LaunchAgaveJob.php [hunk 2]
```diff
@@ -81,10 +82,28 @@ public function handle()
             // foreach ($this->inputs as $key => $value) {
             //     $inputs[$key] = 'agave://' . $this->systemStaging . '/' . $folder . '/' . $value;
             // }
+            Log::debug('#### HELLLOOOOOO');
```
[the remark this answers is not in the document]
Comment (bcorrie) [reply app/Jobs/LaunchAgaveJob.php:85]: 8-) From when I was trying to recall the Laravel framework and how the components worked together. I couldn't figure out where this code was being called from. Will remove. 8-)

[the remark this answers is not in the document]
Comment (bcorrie) [reply app/Jobs/LaunchAgaveJob.php:85]: Fixed...

File: resources/views/sequence.blade.php [hunk 3]
```diff
@@ -6,7 +6,7 @@
 @section('content')
 <div class="container-fluid sequence_container">
 
-	<h1>2. Sequence Search</h1>
+	<h1>2. SSSSSequence Search</h1>
```
[the remark this answers is not in the document]
Comment (bcorrie) [reply resources/views/sequence.blade.php:9]: Nope, intentional - again making sure I new I was changing the correct code from my early explorations. My internal way of knowing I was on the Analysis sequence page 8-)

Shouldn't be part of the end product though - and I don't think I need it any more 8-)

[the remark this answers is not in the document]
Comment (bcorrie) [reply resources/views/sequence.blade.php:9]: Fixed.

File: app/Agave.php [hunk 4]
```diff
@@ -111,6 +112,41 @@ public function renewToken($url, $refresh_token, $api_key, $api_secret)
...
+        foreach ($app_directories as $app_dir) {
+            $app_info = [];
+            $file_path = resource_path('agave_apps/' . $app_dir . '/app.json');
+            Log::debug('updateAppTemplates: Trying to open App file ' . $file_path);
+            $files = scandir('.');
```
[the remark this answers is not in the document]
Comment (bcorrie) [reply app/Agave.php:123]: Yep, was used in a debug statements earlier... fixed...

File: app/Agave.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (bcorrie) [reply app/Agave.php:118]: @jeromejaglale does that make sense...

**Pull request closed without merging**